packages/client/components/BottomControlBarReadyButton.tsx

```diff
@@ -0,0 +1,169 @@
+  }
+  60% {
+    transform: rotate(-7deg);
+  }
+  100% {
+    transform: rotate(0deg);
+  }
+`
+
+const BottomNavReadyButton = styled('div')<{showShakeAnimation?: boolean; delay?: number}>(
+  ({showShakeAnimation, delay}) => {
+    return {
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center',
+      animation: showShakeAnimation ? `1s ease-in-out ${delay}s 3 ${shakeAnimation}` : undefined
+    }
+  }
+)
+
+interface Props {
+  meetingRef: BottomControlBarReadyButton_meeting$key
+  progress: number
+  children: ReactNode
+}
+
+const BottomControlBarReadyButton = (props: Props) => {
+  const {meetingRef, progress, children} = props
+  const meeting = useFragment(
+    graphql`
+      fragment BottomControlBarReadyButton_meeting on NewMeeting {
+        localPhase {
+          phaseType
+        }
+        ... on RetrospectiveMeeting {
+          endedAt
+          facilitatorUserId
+          localPhase {
+            ... on ReflectPhase {
+              reflectPrompts {
+                id
+                editorIds
+              }
+            }
+          }
+          reflectionGroups {
+            id
+            reflections {
+              id
+              isEditing
+              isDropping
+              isViewerDragging
+            }
+          }
+          viewerMeetingMember {
+            votesRemaining
+          }
+          votesRemaining
+        }
+      }
+    `,
+    meetingRef
+  )
+  const {
+    endedAt,
+    facilitatorUserId,
+    localPhase,
+    reflectionGroups,
+    viewerMeetingMember,
+    votesRemaining
+  } = meeting
+  const [delaySeconds, setDelaySeconds] = useState<number>()
+  const atmosphere = useAtmosphere()
+  const {viewerId} = atmosphere
+
+  useEffect(() => {
+    if (localPhase.phaseType !== 'reflect') return
+
+    const hasNoReflection = reflectionGroups?.every((group) => group.reflections?.length === 0)
+    if (hasNoReflection) return
+
+    const reflectPrompts = localPhase.reflectPrompts!
+    const beforeDemoStart =
+      isDemoRoute() && reflectPrompts.every(({editorIds}) => !Array.isArray(editorIds))
+    if (beforeDemoStart) return
+
+    const isNotEditing = reflectPrompts.every(({editorIds}) => {
+      return editorIds === undefined || (Array.isArray(editorIds) && editorIds.length === 0)
+    })
+
+    setDelaySeconds(isNotEditing ? 30 : undefined)
+  }, [localPhase.phaseType === 'reflect', reflectionGroups, localPhase])
+
+  useEffect(() => {
+    if (localPhase.phaseType !== 'group') return
+
+    const isNotDragging = reflectionGroups?.every((group) =>
+      group.reflections?.every(
+        (reflection) => !reflection.isViewerDragging && !reflection.isDropping
+      )
+    )
+
+    setDelaySeconds(isNotDragging ? 30 : undefined)
+  }, [localPhase.phaseType === 'group', reflectionGroups, localPhase])
+
+  useEffect(() => {
+    if (localPhase.phaseType !== 'vote') return
+
+    const teamVotesRemaining = votesRemaining || 0
```

Comment:
-1 `votesRemaining` should be non-null. Also if you check for `null`/`undefined` you should use `??` to make your intention clear, since `!!0 === false`